chapter2/tests/t19/terraform/base/main.tf

```diff
@@ -0,0 +1,27 @@
+module "nginx" {
+  source                       = "../_modules/nginx"
+  container_resource           = "nginx"
+  container_image              = var.nginx.image
+  container_name               = var.nginx.container_name != "" ? var.nginx.container_name : null
+  container_ports              = var.nginx.container_ports != "" ? var.nginx.container_ports : null
+  container_volumes            = var.nginx.container_volumes
+  container_image_keep_locally = var.nginx.keep_locally
+  client                       = var.client
+  project                      = var.project
+  environment                  = var.environment
+}
+
+module "redis" {
+  source                       = "../_modules/container"
+  count                        = var.use_redis ? 1 : 0
+  container_resource           = "redis"
+  container_image              = var.redis.image
+  container_name               = var.use_redis ? var.redis.container_name : null
```

Comment:
no need to use `var.use_redis ?` here, you already check on line 16 with `count`

Reply:
> no need to use `var.use_redis ?` here, you already check on line 16 with `count`

@dmitry-mightydevops 
Done